src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - You can remove the manual assignments of `a`, `b`, and `c` to class properties since TypeScript does this automatically when you use the `public` keyword in the constructor parameters.